docker-multinode/common.sh

```diff
@@ -0,0 +1,437 @@
+  kube::log::status "DNS_DOMAIN is set to: ${DNS_DOMAIN}"
+  kube::log::status "DNS_SERVER_IP is set to: ${DNS_SERVER_IP}"
+  kube::log::status "RESTART_POLICY is set to: ${RESTART_POLICY}"
+  kube::log::status "MASTER_IP is set to: ${MASTER_IP}"
+  kube::log::status "ARCH is set to: ${ARCH}"
+  kube::log::status "--------------------------------------------"
+}
+
+# Detect the OS distro, we support ubuntu, debian, mint, centos, fedora and systemd dist
+kube::multinode::detect_lsb() {
+
+  # TODO(luxas): add support for arm, arm64 and ppc64le
+  case "$(kube::util::host_platform)" in
+    linux/amd64)
+      ;;
+    *)
+      kube::log::error "Error: We currently only support the linux/amd64 platform."
+      exit 1
+      ;;
+  esac
+
+  if kube::helpers::command_exists lsb_release; then
+    lsb_dist="$(lsb_release -si)"
+  elif [[ -r /etc/lsb-release ]]; then
+    lsb_dist="$(. /etc/lsb-release && echo "$DISTRIB_ID")"
+  elif [[ -r /etc/debian_version ]]; then
+    lsb_dist='debian'
+  elif [[ -r /etc/fedora-release ]]; then
+    lsb_dist='fedora'
+  elif [[ -r /etc/os-release ]]; then
+    lsb_dist="$(. /etc/os-release && echo "$ID")"
+  elif kube::helpers::command_exists systemctl; then
+    lsb_dist='systemd'
+  fi
+
+  lsb_dist="$(echo ${lsb_dist} | tr '[:upper:]' '[:lower:]')"
+
+  case "${lsb_dist}" in
+      amzn|centos|debian|ubuntu|systemd)
+          ;;
+      *)
+          kube::log::error "Error: We currently only support ubuntu|debian|amzn|centos|systemd."
+          exit 1
+          ;;
+  esac
+
+  kube::log::status "Detected OS: ${lsb_dist}"
+}
+
+# Start a docker bootstrap for running etcd and flannel
+kube::multinode::bootstrap_daemon() {
+
+  kube::log::status "Launching docker bootstrap..."
+
+  docker daemon \
+    -H ${BOOTSTRAP_DOCKER_SOCK} \
+    -p /var/run/docker-bootstrap.pid \
+    --iptables=false \
+    --ip-masq=false \
+    --bridge=none \
+    --graph=/var/lib/docker-bootstrap \
+      2> /var/log/docker-bootstrap.log \
+      1> /dev/null &
+
+  # Wait for docker bootstrap to start by "docker ps"-ing every second
+  local BOOTSTRAP_SECONDS=0
+  while [[ $(docker -H ${BOOTSTRAP_DOCKER_SOCK} ps 2>&1 1>/dev/null; echo $?) != 0 ]]; do
+    ((BOOTSTRAP_SECONDS++))
+    if [[ ${BOOTSTRAP_SECONDS} == ${TIMEOUT_FOR_SERVICES} ]]; then
+      kube::log::error "docker bootstrap failed to start. Exiting..."
+      exit
+    fi
+    sleep 1
+  done
+}
+
+# Start etcd on the master node
+kube::multinode::start_etcd() {
+
+  kube::log::status "Launching etcd..."
+  
+  docker -H ${BOOTSTRAP_DOCKER_SOCK} run -d \
+    --restart=${RESTART_POLICY} \
+    --net=host \
+    gcr.io/google_containers/etcd-${ARCH}:${ETCD_VERSION} \
+    /usr/local/bin/etcd \
+      --listen-client-urls=http://127.0.0.1:4001,http://${MASTER_IP}:4001 \
+      --advertise-client-urls=http://${MASTER_IP}:4001 \
+      --data-dir=/var/etcd/data
+
+  # Wait for etcd to come up
+  kube::util::wait_for_url "http://localhost:4001/v2/machines" "etcd" 0.25 80
```

Comment:
this wait is not working. etcd returns some unexpected value

Adding sleep 5 fixes the script